core/src/main/java/com/homeaway/streamingplatform/db/dao/impl/ProducerDaoImpl.java

```diff
@@ -148,16 +147,20 @@ private void deleteProducer(String streamName, String producerName) {
         Optional<AvroStream> avroStream = getAvroStreamKeyValue(streamName).getValue();
 
         if (avroStream.isPresent()) {
-            List<com.homeaway.digitalplatform.streamregistry.Producer> listProducer = avroStream.get().getProducers();
-            int producerInitialSize = listProducer.size();
-            for (Iterator<com.homeaway.digitalplatform.streamregistry.Producer> iter = listProducer.listIterator(); iter.hasNext(); ) {
-                com.homeaway.digitalplatform.streamregistry.Producer producerEntity = iter.next();
-                if (producerEntity.getActor().getName().equalsIgnoreCase(producerName)) {
-                    // Remove the managedKafkaProducer
-                    iter.remove();
-                    avroStream.get().setProducers(Lists.newArrayList(iter));
-                }
-            }
+            // Obtains producer list size before  remove consumer
+            final int producerInitialSize = avroStream.get().getProducers().size();
+
+            // Obtains filtered producer list not containing the consumer we want to remove
+            List<com.homeaway.digitalplatform.streamregistry.Producer> withoutProducer = avroStream.get()
+                    .getProducers()
```

Comment:
`avroStream.get().getProducers()` can be refactored to a variable since it's used twice

Reply:
It's ready